assets/src/models/createDetourMachine.ts

```diff
@@ -282,6 +303,22 @@ export const createDetourMachine = setup({
               },
             },
             "Finished Drawing": {
+              invoke: {
+                src: "fetch-finished-detour",
+                input: ({
+                  context: { routePattern, startPoint, endPoint },
+                }) => ({
+                  routePatternId: routePattern?.id,
+                  startPoint,
+                  endPoint,
+                }),
+
+                onDone: {
+                  actions: assign({
+                    finishedDetour: ({ event }) => event.output,
+                  }),
+                },
```

Comment:
issue: btw, if you don't stub out the `onError` handler, the state machine will lockup if the API call fails in _some_ way. So similar to the `fetch-route-patterns` actor, you should add an empty handler.
https://github.com/mbta/skate/blob/jdl/refactor/finished-detour-into-state-machine/assets/src/models/createDetourMachine.ts#L180-L181

Reply:
Hmm, actually I'm not to sure on this anymore, I had _thought_ that this was the case, but lets check this assumption while pairing this afternoon.